### ISSUE TITLE
Paid partnership with

### ISSUE THREAD

**Issue by kooshazarei** [instaloader/instaloader#295]
Hi everyone,

I have a question, I would like to get "Paid partnership with" metadata of Instagram posts. As far as I looked there is no option in Instaloader. is it right? if there is, how can I get this metadata in JSON?

I look forward to hearing from you,
Thanks

**Comment by GolvenLD** [comment]
Is this information usually in the caption ? If it is (and I have no idea) you can get the caption and parse it to get only what you want? I just don't know on Instagram where this metadata is.

**Comment by kooshazarei** [comment]
> Is this information usually in the caption ? If it is (and I have no idea) you can get the caption and parse it to get only what you want? I just don't know on Instagram where this metadata is.

on Instagram, this information is shown in a post location, but I couldn't find in the Instaloader.

**Comment by aandergr** [comment]
Could you show us an example post that has this information on Instagram?

**Comment by kooshazarei** [comment]
> Could you show us an example post that has this information on Instagram?

Yes, of course, check these out:

-https://later.com/blog/paid-partnership-feature/
-https://www.agorapulse.com/blog/instagram-influencers

**Comment by BW0ng** [comment]
> Could you show us an example post that has this information on Instagram?

https://www.instagram.com/p/Bx201VODYoW/
https://www.instagram.com/p/By-pItMhM-t/
https://www.instagram.com/p/ByV3RlRBS3k/

Here's a couple posts as well as the official [page ](https://business.instagram.com/a/brandedcontentexpansion) from instagram

**Comment by e5150** [comment]
After a quick look, it seems already be available in the json stored by Instaloader.
`xzcat *.json.xz | jq '.node.edge_media_to_sponsor_user.edges[0].node.sponsor.username'`

**Comment by aandergr** [comment]
We implemented this feature with one of our latest commits, and it will be included in the upcoming version, Instaloader 4.4. To already try it out, upgrade Instaloader with:

```
pip3 install --upgrade git+https://github.com/instaloader/instaloader@v4.4-dev
```
